src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -30,12 +30,21 @@ public Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(C
                 // TODO: return empty outside of debug session.
                 // https://github.com/dotnet/roslyn/issues/24325
 
-                var workList = DkmWorkList.Create(CompletionRoutine: null);
+                bool completed = false;
+                int unexpectedError = 0;
                 var completion = new TaskCompletionSource<ImmutableArray<ActiveStatementDebugInfo>>();
                 var builders = default(ArrayBuilder<ArrayBuilder<ActiveStatementDebugInfo>>);
                 int pendingRuntimes = 0;
                 int runtimeCount = 0;
 
+                var workList = DkmWorkList.Create(CompletionRoutine: _ =>
+                {
+                    if (!completed)
+                    {
+                        completion.TrySetException(new InvalidOperationException($"Unexpected error enumerating active statements: 0x{unexpectedError:X8}"));
```

Comment:
Since you're already using the Try* stuff, why have the completed flag?

Reply:
Does this do nothing if it's already completed in other way?

Reply:
Correct. Set* methods will throw if any Set* method was already called, TrySet* does nothing. TBH, we tend to always use TrySet in Roslyn because it often simplifies code and makes things more reliable too.